File: Exercitar java/Exercicio05/Exercicio05.java






class Main{
    
    //Função para verificar se um número é primo
    static boolean serPrimo(int numero){

        //Verificação
        for(int i = 2; i < numero; i++){
            if(numero % i == 0){
                return false;
            }
            
        }
        return true;
    }
    
    
    public static void main(String[] args) throws Exception {
    
        //Leitura do número 
        System.out.println("Insira o numero :");
        int numero = EntradaTeclado.leInt();

        //Declaração das variavéis utilizadas
        int numeroAlvo = numero-1; 
        boolean alvoPrimo = false; //Verifica se o número alvo atual é primo
        
        while(alvoPrimo == false && numeroAlvo > 1){
            numeroAlvo--;            
            alvoPrimo = serPrimo(numeroAlvo);
        }
        if(numeroAlvo == 1){
            System.out.printf("Não existe número primo menor que %d\n",numero);
        }
        if(alvoPrimo == true){
            System.out.printf("O primeiro numero primo menor que %d é %d\n",numero, numeroAlvo);
        }
    }
}

File: Exercitar python/Exercicio05.py



def verPrimo(numero):
    
    for i in range(2,numero):
        if(numero % i == 0):
            return False
    return True

def main():

    print("Insira um número inteiro")
    numero = int(input())

    i = numero - 1
    while(i > 1):
        if(verPrimo(i)):
            print("O menor número primo é ",i)
            quit()

        i -= 1
    print("Não existe numero primo menor que ",numero)  
    

if __name__ == "__main__":
    main()

